### PR TITLE
New scripts: isUrlPartOfDomain

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/26220

## Minimum version of Demisto
- [x] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Must have
- [x] Tests

### CODE REVIEW

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  id: IsUrlPartOfDomain
+  version: -1
+enabled: false
+name: IsUrlPartOfDomain
+outputs:
+- contextPath: IsUrlPartOfDomain.URL
+  description: The path of the URLs.
+  type: String
+- contextPath: IsUrlPartOfDomain.Domain
+  description: The path of the URLs.
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml:25]: fix output descriptions

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    Args:
+        domain_name: A domain
+        urls: URLs to check if the domain in them.
+
+    Returns:
+        Results to display in CortexXSOAR
+    """
+    urls = argToList(urls)
+    results = demisto.executeCommand('ExtractDomainFromUrlAndEmail', {'input': domain_name})
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:17]: check for error with `is_error` and use `get_error` to extract it

Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:17]: why are we even running that ? why not just run it for each url ?

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    domain = results[0]['Contents']
+
+    if not domain:
+        return_error(f'Could not find a domain in "{domain_name}"')
+    domain = domain.lower()
+
+    outputs: list = list()
+    for url in urls:
+        results = demisto.executeCommand('ExtractDomainFromUrlAndEmail', {'input': url.lower()})
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:26]: dont run the script for each, just pass the csv

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  type: String
+- contextPath: IsUrlPartOfDomain.IsInternal
+  description: Is the url is the domain.
+  type: Boolean
+script: '-'
+subtype: python3
+system: false
+timeout: '0'
+type: python
+dockerimage: demisto/tld:1.0.0.8854
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml:35]: why do you need that docker image?

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py (hunk 3 above)
Comment (jochman) [reply Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:26]: But then I'll get multiple domains so it's open to bugs.

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py (hunk 2 above)
Comment (jochman) [reply Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:17]: We're running it for each url.

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            demisto.debug(f'Could not get domain from url {url}')
+            return_warning(get_error(results))
+        else:
+            domain_from_url = results[0]['Contents']
+            outputs.append({
+                'URL': url,
+                'Domain': domain,
+                'IsInternal': domain == domain_from_url
+            })
+    return CommandResults('IsUrlPartOfDomain', 'Data', outputs)
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:34]: why is `Data` the key here?

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    """
+    urls = argToList(urls)
+    domain = domain_name.lower()
+
+    outputs: List[Dict] = list()
+    for url in urls:
+        results = demisto.executeCommand('ExtractDomainFromUrlAndEmail', {'input': url.lower()})
+        if is_error(results):
+            demisto.debug(f'Could not get domain from url {url}')
+            return_warning(get_error(results))
```
Comment (Itay4) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.py:26]: lets raise an exception / return error here, that means the script failed to run

File: Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+args:
+- default: false
+  description: Name of the domain
+  isArray: false
+  name: domain_name
```
Comment (David-BMS) [inline Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml:5]: Needs to be able to handle a comma delimited domain list

Comment (jochman) [reply Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml]: Done 👍

Comment (Itay4) [reply Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml]: @jochman please add test for that

Comment (jochman) [reply Packs/CommonScripts/Scripts/IsUrlPartOfDomain/IsUrlPartOfDomain.yml]: pushed

**Comment by kirbles19** [comment]
@jochman Done.

**Pull request merged**